runbot/runbot.py

```diff
@@ -1243,11 +1244,27 @@ def branch_info(branch):
 
         return request.render("runbot.repo", context)
 
-    @http.route(['/runbot/hook/<int:repo_id>'], type='http', auth="public", website=True)
+    @http.route(['/runbot/hook/<int:repo_id>', '/runbot/hook/org'], type='json', auth="public", website=True)
     def hook(self, repo_id=None, **post):
-        # TODO if repo_id == None parse the json['repository']['ssh_url'] and find the right repo
-        repo = request.registry['runbot.repo'].browse(request.cr, SUPERUSER_ID, [repo_id])
-        repo.hook_time = datetime.datetime.now().strftime(openerp.tools.DEFAULT_SERVER_DATETIME_FORMAT)
+        if repo_id is None:
+            repo_data = request.jsonrequest.get('repository')
+            event = request.httprequest.headers.get("X-Github-Event")
+            if repo_data and event in ['push', 'pull_request']:
+                repo_owner = repo_data['owner'].get('name') or repo_data['owner'].get('login')
+                repo_name = repo_data['name']
```

Comment:
Instead of parsing `repo.owner.name/login` and `repo.name`, why don't you use [`repo.full_name`](https://developer.github.com/v3/activity/events/types/#repositoryevent) directly?